src/Data/URI.php

```diff
@@ -381,4 +381,56 @@ public function getBaseURI() : string
         }
         return $this->getSchema() . '://' . $this->getAuthority() . '/' . $path;
     }
+
+    public function __toString() : string
+    {
+        $uri = $this->getBaseURI() . '?' . $this->getQuery();
+        if ($this->fragment) {
+            $uri .= '#' . $this->getFragment();
+        }
+        return $uri;
+    }
+
+    /**
+     * Get all parameters as associative array
+     */
+    public function getParameters() : array
+    {
+        $params = [];
+        parse_str($this->getQuery(), $params);
+        return $params;
+    }
+
+    /**
+     * Get the value of the given parameter (or null)
+     * @return  mixed|null
+     */
+    public function getParameter(string $param)
```

Comment:
You could use `:? string` as explicit return type here.

Reply:
Hi @mjansenDatabay , 
I did, at first, but arrays do work :)
please see [here](https://github.com/ILIAS-eLearning/ILIAS/pull/2864/files#diff-71a136a2582520363a4a814f20782245R945-R948).